beacon_node/lighthouse_network/src/types/topics.rs

```diff
@@ -23,21 +23,47 @@ pub const BLS_TO_EXECUTION_CHANGE_TOPIC: &str = "bls_to_execution_change";
 pub const LIGHT_CLIENT_FINALITY_UPDATE: &str = "light_client_finality_update";
 pub const LIGHT_CLIENT_OPTIMISTIC_UPDATE: &str = "light_client_optimistic_update";
 
-pub const CORE_TOPICS: [GossipKind; 7] = [
+pub const BASE_CORE_TOPICS: [GossipKind; 5] = [
     GossipKind::BeaconBlock,
     GossipKind::BeaconAggregateAndProof,
     GossipKind::VoluntaryExit,
     GossipKind::ProposerSlashing,
     GossipKind::AttesterSlashing,
-    GossipKind::SignedContributionAndProof,
-    GossipKind::BlsToExecutionChange,
 ];
 
+pub const ALTAIR_CORE_TOPICS: [GossipKind; 1] = [GossipKind::SignedContributionAndProof];
+
+pub const CAPELLA_CORE_TOPICS: [GossipKind; 1] = [GossipKind::BlsToExecutionChange];
+
+pub const EIP4844_CORE_TOPICS: [GossipKind; 1] = [GossipKind::BeaconBlocksAndBlobsSidecar];
+
 pub const LIGHT_CLIENT_GOSSIP_TOPICS: [GossipKind; 2] = [
     GossipKind::LightClientFinalityUpdate,
     GossipKind::LightClientOptimisticUpdate,
 ];
 
+/// Returns the core topics associated with each fork
+pub fn fork_core_topics(fork_name: &ForkName) -> Vec<GossipKind> {
+    match fork_name {
+        ForkName::Base => BASE_CORE_TOPICS.to_vec(),
+        ForkName::Altair => ALTAIR_CORE_TOPICS.to_vec(),
+        ForkName::Merge => vec![],
+        ForkName::Capella => CAPELLA_CORE_TOPICS.to_vec(),
+        ForkName::Eip4844 => EIP4844_CORE_TOPICS.to_vec(),
+    }
+}
+
+/// Returns all the topics that we need to subscribe to for a given fork
+/// including topics from older forks and new topics for the current fork.
+pub fn core_topics_to_subscribe(current_fork: ForkName) -> Vec<GossipKind> {
+    let mut topics = fork_core_topics(&current_fork);
+    while let Some(previous_fork) = current_fork.previous_fork() {
+        let mut previous_fork_topics = fork_core_topics(&previous_fork);
+        topics.append(&mut previous_fork_topics);
+    }
+    topics
+}
```

Comment:
very elegant

Reply:
damn, i had fixed it after running a local testnet. Didn't commit. Thanks for catching it.
Added a small test for the function now.